python/chemiscope/structures/__init__.py

```diff
@@ -65,7 +66,10 @@ def frames_to_json(frames):
     elif adapter == "mda":
         # Be careful of the lazy loading of `frames.atoms`, which is updated during the
         # iteration of the trajectory
-        return [_mda_to_json(frames) for _ in frames.universe.trajectory]
+        data_collection = [_mda_to_json(frames) for _ in frames.universe.trajectory]
+        if all("residues" in data for data in data_collection):
+            data_collection = _mda_get_secondary_structure(data_collection, frames)
```

Comment:
this should ideally be done by `_mda_to_json`, the functions in this file should not have to care